Content.Server/_Moffstation/GameTicking/Rules/GunGameRuleSystem.cs

```diff
@@ -98,18 +98,22 @@ private void OnKillReported(ref KillReportedEvent ev)
         var query = EntityQueryEnumerator<GunGameRuleComponent, RespawnTrackerComponent, GameRuleComponent>();
         while (query.MoveNext(out var uid, out var gunGame, out var respawnTracker, out var rule))
         {
-            if (!GameTicker.IsGameRuleActive(uid, rule)
-                || ev.Primary is not KillPlayerSource player)
+            if (!GameTicker.IsGameRuleActive(uid, rule))
+		continue;
+
+            // Force them to respawn so they don't have to wait around.
+            // This has to happen before we check if a player killed them, because they might have died
+            // 	to something other than a player.
+            if (TryComp<ActorComponent>(ev.Entity, out var actor))
+                _respawn.RespawnPlayer((ev.Entity, actor), (uid, respawnTracker));
+
+            if (ev.Primary is not KillPlayerSource player)
                 continue;
 
             // Don't want other players picking up somebody's gun
             if (TryComp<GunGameTrackerComponent>(ev.Entity, out var gunGameTracker))
                 DeleteCurrentWeapons((ev.Entity, gunGameTracker), gunGame);
```

Comment:
Still need to delete all their things so they can't be stolen.

```suggestion
            // Don't want other players picking up somebody's gun
            if (TryComp<GunGameTrackerComponent>(ev.Entity, out var gunGameTracker))
                DeleteCurrentWeapons((ev.Entity, gunGameTracker), gunGame);
                
            if (ev.Primary is not KillPlayerSource player)
                continue;
```

Reply:
Ah, right, will do.